lib/MongoDB/Collection.pm

```diff
@@ -1857,6 +1857,9 @@ sub __ixhash {
     elsif ( $type eq 'ARRAY' ) {
         $hash->{$key} = Tie::IxHash->new( @$ref );
     }
+    elsif ( $ref->$_can('_as_tied_hash') ) {
```

Comment:
Problem with this is that even though it works, it's not an technically a Tie::IxHash. (I'm sure later code converts it.)  I think you can just inline what `__as_tied_hash` does, ie. dereferencing the object to a list of key/value pairs to pass to the IxHash constructor.  Doing that is no worse than using the private method.  :-)

Reply:
Yea later code does convert it - putting `return if $type eq 'BSON::Doc';` works as well... have added it as a logical or to the ARRAY type, as its identical conversion.